src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
you'd start with an initial guess, maybe ones(d), and repeatedly multiply it by the covariance matrix from summing up the outer products of all the vectors, which is the same as adding up the dot product of the guess with the outer product of each vector with itself... here: http://www.texpaste.com/n/7ao61x20